py/tests/list.py

```diff
@@ -35,6 +35,13 @@
 assert repr(a) == "['a', 'b', 'c', 'd', 'e', 'f']"
 assertRaises(TypeError, lambda: [].append())
 
+doc="pop"
+a = [1,2,3,4]
+assert a.pop() == 1
+assert a.pop(2) == 4
```

Comment:
The `pop` operation removes and returns from the back, I think your idea was removing from the front.
It would be good if you create a new list for each test and compare your test against a list like `assert a.pop(-1) == elemtent; assert a = [...]`, that makes it more clear here.

Additionally the following tests would be good:
* negative index
* index out of range for non-empty list
* parameter of wrong type (non-int)